addons/sale/models/sale_order_line.py

```diff
@@ -365,29 +365,23 @@ def _get_delivered_quantity_by_analytic(self, additional_domain):
 
         # group analytic lines by product uom and so line
         domain = expression.AND([[('so_line', 'in', self.ids)], additional_domain])
-        data = self.env['account.analytic.line'].read_group(
-            domain,
-            ['so_line', 'unit_amount', 'product_uom_id'], ['product_uom_id', 'so_line'], lazy=False
-        )
-
-        # convert uom and sum all unit_amount of analytic lines to get the delivered qty of SO lines
-        # browse so lines and product uoms here to make them share the same prefetch
-        lines = self.browse([item['so_line'][0] for item in data])
-        lines_map = {line.id: line for line in lines}
-        product_uom_ids = [item['product_uom_id'][0] for item in data if item['product_uom_id']]
-        product_uom_map = {uom.id: uom for uom in self.env['uom.uom'].browse(product_uom_ids)}
-        for item in data:
-            if not item['product_uom_id']:
+        analytic_lines = self.env['account.analytic.line'].search(domain)
+        for line in analytic_lines:
+            sign = 1
+            if not line.product_uom_id:
                 continue
-            so_line_id = item['so_line'][0]
-            so_line = lines_map[so_line_id]
-            result.setdefault(so_line_id, 0.0)
-            uom = product_uom_map.get(item['product_uom_id'][0])
+            so_line = line.so_line
+            result.setdefault(so_line.id, 0.0)
+            uom = line.product_uom_id
```

Comment:
IMO using an intermediate variable makes the code less readable in this case

Reply:
indeed